web/src/js/__tests__/ducks/modes/regularSpec.tsx

```diff
@@ -0,0 +1,123 @@
+                    value: ["regular@http:8081"],
+                },
+            },
+        };
+        const newState = regularReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.listen_host).toBe("http");
+        expect(newState.listen_port).toBe(8081);
+    });
+
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "regular" and a port', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["regular@8081"],
+                },
+            },
+        };
+        const newState = regularReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.listen_host).toBe(undefined);
+        expect(newState.listen_port).toBe(8081);
+    });
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "regular"', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["regular"],
+                },
+            },
+        };
+        const newState = regularReducer(initialState, action);
+        expect(newState.active).toBe(true);
+        expect(newState.listen_host).toBe(undefined);
+        expect(newState.listen_port).toBe(undefined);
+    });
+});
+
+describe("getMode", () => {
+    it("should return the correct mode string when active", () => {
+        const modes = {
+            regular: {
+                active: true,
+                name: "regular",
+                listen_host: "localhost",
+                listen_port: 8080,
+            },
+        };
+        const mode = getMode(modes);
+        expect(JSON.stringify(mode)).toBe(
+            JSON.stringify(["regular@localhost:8080"]),
+        );
+    });
+
+    it("should return an empty string when not active", () => {
+        const modes = {
+            regular: {
+                active: false,
+                listen_host: "localhost",
+                listen_port: 8080,
+            },
+        };
+        const mode = getMode(modes);
+        expect(JSON.stringify(mode)).toBe(JSON.stringify([]));
+    });
+
+    it("should return the correct mode string without listen_host and listen_port", () => {
+        const modes = {
+            regular: {
+                active: true,
+                name: "regular",
+            },
+        };
+        const mode = getMode(modes);
+        expect(JSON.stringify(mode)).toBe(JSON.stringify(["regular"]));
+    });
+
+    it("should return the correct mode string with listen_port and without listen_host", () => {
+        const modes = {
+            regular: {
+                active: true,
+                name: "regular",
+                listen_port: 8080,
+            },
+        };
+        const mode = getMode(modes);
+        expect(JSON.stringify(mode)).toBe(JSON.stringify(["regular@8080"]));
+    });
```

Comment:
Two tests - one for active, one for not active is good enough here as long as we test addListenAddr separately at one central place. :)